File: server/models/requirements.py
from pydantic import BaseModel, Field
from typing import List


class FunctionalRequirement(BaseModel):
    name: str
    description: str
    priority: str


class NonFunctionalRequirement(BaseModel):
    name: str
    description: str
    priority: str


class Requirements(BaseModel):
    functional_requirements: List[FunctionalRequirement] = Field(default_factory=list)
    non_functional_requirements: List[NonFunctionalRequirement] = Field(
        default_factory=list
    )

File: client2/src/model/requirements.ts
export type FunctionalRequirement = {
    name: string;
    description: string;
    priority: string;
};

export type NonFunctionalRequirement = {
    name: string;
    description: string;
    priority: string;
};

export type Requirements = {
    functional_requirements: FunctionalRequirement[];
    non_functional_requirements: NonFunctionalRequirement[];
};
